istio-telemetry/kiali-operator/values.yaml

```diff
@@ -0,0 +1,62 @@
+#
+# addon kiali
+#
+kiali:
+  replicaCount: 1
+  hub: docker.io/kiali
+  tag: v1.1.0
+  contextPath: /kiali # The root context path to access the Kiali UI.
+  nodeSelector: {}
+
+  # Specify the pod anti-affinity that allows you to constrain which nodes
+  # your pod is eligible to be scheduled based on labels on pods that are
+  # already running on the node rather than based on labels on nodes.
+  # There are currently two types of anti-affinity:
+  #    "requiredDuringSchedulingIgnoredDuringExecution"
+  #    "preferredDuringSchedulingIgnoredDuringExecution"
+  # which denote “hard” vs. “soft” requirements, you can define your values
+  # in "podAntiAffinityLabelSelector" and "podAntiAffinityTermLabelSelector"
+  # correspondingly.
+  # For example:
+  # podAntiAffinityLabelSelector:
+  # - key: security
+  #   operator: In
+  #   values: S1,S2
+  #   topologyKey: "kubernetes.io/hostname"
+  # This pod anti-affinity rule says that the pod requires not to be scheduled
+  # onto a node if that node is already running a pod with label having key
+  # “security” and value “S1”.
+  podAntiAffinityLabelSelector: []
+  podAntiAffinityTermLabelSelector: []
+
+  ingress:
+    enabled: false
+    ## Used to create an Ingress record.
+    hosts:
+      - kiali.local
+    annotations:
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
+    tls:
+    # Secrets must be manually created in the namespace.
+    # - secretName: kiali-tls
+    #   hosts:
+    #     - kiali.local
+
+  dashboard:
+    auth:
+      strategy: login # Can be anonymous, login, or openshift
+    secretName: kiali # You must create a secret with this name - one is not provided out-of-box.
+    viewOnlyMode: false # Bind the service account to a role with only read access
+    grafanaURL:  # If you have Grafana installed and it is accessible to client browsers, then set this to its external URL. Kiali will redirect users to this URL when Grafana metrics are to be shown.
+    jaegerURL:  # If you have Jaeger installed and it is accessible to client browsers, then set this property to its external URL. Kiali will redirect users to this URL when Jaeger tracing is to be shown.
+
+  prometheusAddr: http://prometheus:9090
```

Comment:
I assume this is not always the case when kiali is not in the same namespace as prometheus.

Is it possible to remove the need of `istio_namespace: {{ .Release.Namespace }}` in kiali.yaml and have user to explicitly specify the address with namespace in it if needed?

Reply:
I think it is doable for grafana/jaeger/prometheus in chart. but it requires code change in kiali side - https://github.com/istio/installer/pull/295#discussion_r309297197